### PR TITLE
fix(feetiers): Reject malformed address in query

### PR DESCRIPTION
**Pull request by teddyding**
### Changelist
https://dydx-team.slack.com/archives/C03FVKBHD5H/p1732319479334399

### Test Plan
Unit test

### Author/Reviewer Checklist
- [ ] If this PR has changes that result in a different app state given the same prior state and transaction list, manually add the `state-breaking` label.
- [ ] If the PR has breaking postgres changes to the indexer add the `indexer-postgres-breaking` label.
- [ ] If this PR isn't state-breaking but has changes that modify behavior in `PrepareProposal` or `ProcessProposal`, manually add the label `proposal-breaking`.
- [ ] If this PR is one of many that implement a specific feature, manually label them all `feature:[feature-name]`.
- [ ] If you wish to for mergify-bot to automatically create a PR to backport your change to a release branch, manually add the label `backport/[branch-name]`.
- [ ] Manually add any of the following labels: `refactor`, `chore`, `bug`.

## Summary by CodeRabbit

- **New Features**
	- Enhanced user address validation in the fee tier request process.

- **Bug Fixes**
	- Improved error handling for invalid user addresses.

- **Tests**
	- Updated test cases to include validation for valid and malformed user addresses.

### CODE REVIEW

File: protocol/x/feetiers/keeper/grpc_query_test.go [hunk 1]
```diff
@@ -78,6 +78,13 @@ func TestUserFeeTier(t *testing.T) {
 			res: nil,
 			err: status.Error(codes.InvalidArgument, "invalid request"),
 		},
+		"Malformed address": {
+			req: &types.QueryUserFeeTierRequest{
+				User: "alice",
+			},
+			res: nil,
+			err: status.Error(codes.InvalidArgument, "user address is valid bech32 address"),
```
Comment (vincentwschau) [inline protocol/x/feetiers/keeper/grpc_query_test.go:86]: Shouldn't the error message be saying the address is invalid?

**Review by vincentwschau — APPROVED** [review]

**Comment by teddyding** [comment]
@Mergifyio backport release/protcool/v8.x

**Comment by teddyding** [comment]
@Mergifyio backport release/protocol/v8.x

**Comment by teddyding** [comment]
@Mergifyio backport release/protocol/v7.x

**Review by vincentwschau — APPROVED** [review]

**Pull request merged**